OADP-4832 - Release notes for OADP 1.4.2 (#87456)
openshift/openshift-docs: modules/oadp-1-4-2-release-notes.adoc

// Module included in the following assemblies:
//
// * backup_and_restore/oadp-1-4-release-notes.adoc

:_mod-docs-content-type: REFERENCE

[id="oadp-1-4-2-release-notes_{context}"]

> = OADP 1.4.2 release notes

Reviewer: ```suggestion
= {oadp-short} 1.4.2 release notes
```

Author: fixed

The {oadp-first} 1.4.2 release notes lists new features, resolved issues and bugs, and known issues.

[id="new-features-1-4-2_{context}"]
== New features

.Backing up different volumes in the same namespace by using the VolumePolicy feature is now possible

> With this release, Velero now provides resource policies to back up different volumes in the same namespace by using the `VolumePolicy` feature. The supported `VolumePolicy` feature to back up different volumes includes `skip`, `snapshot`, and `fs-backup` actions.

Reviewer: ```suggestion
With this release, Velero provides resource policies to back up different volumes in the same namespace by using the `VolumePolicy` feature. The supported `VolumePolicy` feature to back up different volumes includes `skip`, `snapshot`, and `fs-backup` actions.
```

link:https://issues.redhat.com/browse/OADP-1071[OADP-1071]

.File system backup and data mover can now use short-term credentials

> File system backup and data mover can now use short-term credentials such as {aws-short} {sts-first} and {gcp-short} WIF. With this support, backup successfully completes without any `PartiallyFailed` status.

Reviewer: ```suggestion
File system backup and data mover can now use short-term credentials such as {aws-short} {sts-first} and {gcp-short} WIF. With this support, backup is successfully completed without any `PartiallyFailed` status.
```
In this case, passive voice is acceptable because the system is performing the act. That is, the system is completing the backup. For examples of where the passive voice is acceptable, see the ISG: https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#voice

link:https://issues.redhat.com/browse/OADP-5095[OADP-5095]

[id="resolved-issues-1-4-2_{context}"]
== Resolved issues

.DPA now reports errors if VSL contains an incorrect provider value 

Previously, if the provider of a Volume Snapshot Location (VSL) spec was incorrect, the Data Protection Application (DPA) reconciled successfully. With this update, DPA reports errors and requests for a valid provider value.
link:https://issues.redhat.com/browse/OADP-5044[OADP-5044]

.Data Mover restore is successful irrespective of using different OADP namespaces for backup and restore

Previously, when backup operation was executed by using {oadp-short} installed in one namespace but was restored by using {oadp-short} installed in a different namespace, the Data Mover restore failed. With this update, Data Mover restore is now successful.
link:https://issues.redhat.com/browse/OADP-5460[OADP-5460]

.SSE-C backup works with the calculated MD5 of the secret key

Previously, backup failed with the following error:
[source,terminal]
----
Requests specifying Server Side Encryption with Customer provided keys must provide the client calculated MD5 of the secret key.
----

> With this update, missing Server-Side Encryption with Customer-Provided Keys (SSE-C) base64 and MD5 hash are now fixed. As a result, SSE-C backup works with the calculated MD5 of the secret key. Along with this, incorrect `errorhandling` for the `customerKey` size is also fixed.

Reviewer: ```suggestion
With this update, missing Server-Side Encryption with Customer-Provided Keys (SSE-C) base64 and MD5 hash are now fixed. As a result, SSE-C backup works with the calculated MD5 of the secret key. In addition, incorrect `errorhandling` for the `customerKey` size is also fixed.
```
Rewrite to avoid a potentially ambiguous pronoun ("this" with an unclear noun reference).

link:https://issues.redhat.com/browse/OADP-5388[OADP-5388]

For a complete list of all issues resolved in this release, see the list of link:https://issues.redhat.com/issues/?filter=12452476[OADP 1.4.2 resolved issues] in Jira.

[id="known-issues-1-4-2_{context}"]
== Known issues

.The nodeSelector spec is not supported for the Data Mover restore action

> When a Data Protection Application (DPA) is created with the `nodeSelector` field set in the `nodeAgent` parameter, Data Mover restore partially fails instead of completing the restore operation. 

Reviewer: ```suggestion
When a Data Protection Application (DPA) is created with the `nodeSelector` field set in the `nodeAgent` parameter, Data Mover restore fails instead of completing the restore operation. 
```
Forgive me for being too picky, but is there a difference between "partially fails" and "fails"? I assume that if something partially fails, it fails.

Author: No worries. It does not fail completely and that's how the engineering team refers to it.
There will be cases, where "PartiallyFailed" status will be displayed

link:https://issues.redhat.com/browse/OADP-5260[OADP-5260]

> .The S3 storage does not use proxy environment when tls skip verify is specified

Reviewer: ```suggestion
.The S3 storage does not use proxy environment when TLS skip verify is specified
```

In the image registry backup, the S3 storage does not use the proxy environment when the `insecureSkipTLSVerify` parameter is set to `true`.
link:https://issues.redhat.com/browse/OADP-3143[OADP-3143]

> .Kopia does not delete the artifacts after backup expiration

Reviewer: ```suggestion
.Kopia does not delete artifacts after backup expiration
```

> Even after deleting a backup, Kopia does not delete the volume artifacts from the `${bucket_name}/kopia/${namespace}` on the S3 location after backup expired. 

Reviewer: I'm not sure who is deleting the backup in this case. Is it the user? Is it Kopia? Is it the system?

Depending on who deletes the backup, you might use one of these revisions:
- "Even after you delete a backup, Kopia does not delete the volume artifacts from the `${bucket_name}/kopia/${namespace}` on the S3 location after the backup expired."
- "Even after the backup is deleted, Kopia does not delete the volume artifacts from the `${bucket_name}/kopia/${namespace}` on the S3 location after the backup expired."
- "Even after it deletes the backup, Kopia does not delete the volume artifacts from the `${bucket_name}/kopia/${namespace}` on the S3 location after the backup expired."

Author: Thanks for that, I've fixed it.

> For more information, see link:https://docs.openshift.com/container-platform/4.17/backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-deleting-backups.html#oadp-about-kopia-repo-maintenance_deleting-backups[About Kopia repository maintenance].

Reviewer: ```suggestion
For more information, see "About Kopia repository maintenance".
```
Remove the link, place double-quotation marks around the reference, and add an xref to the reference in an "Additional resources" section in the assembly.

Doing this will also resolve the issue of having two links directly next to each other on the page.

Author: @lahinson, thanks for this suggestion. I've fixed it

// Module included in the following assemblies:
//
// * backup_and_restore/oadp-1-4-release-notes.adoc

:_mod-docs-content-type: REFERENCE

[id="oadp-1-4-2-release-notes_{context}"]
= OADP 1.4.2 release notes

The {oadp-first} 1.4.2 release notes lists new features, resolved issues and bugs, and known issues.

[id="new-features-1-4-2_{context}"]
== New features

> .Backing up different volumes in the same namespace using VolumePolicy feature is now possible

Author: I agree, I've fixed it.
I was trying to keep the release note title short and explain it in the main text :)

> With this release, Velero now provides resource policies to backup different volumes in the same namespace through the `VolumePolicy` feature. The supported `VolumePolicy` feature to backup different volumes include `skip`, `snapshot`, and `fs-backup` actions.

Reviewer 2: ```suggestion
With this release, Velero now provides resource policies to back up different volumes in the same namespace through the `VolumePolicy` feature. The supported `VolumePolicy` feature to back up different volumes includes `skip`, `snapshot`, and `fs-backup` actions.
```

* `backup` is a noun
* `back up` is a verb phrase

Reviewer 3: - I'd also say that "through" should be replaced with something different. IBM SG warns us about its usage. What about repeating ourselves and replace it with "by using" or "with"?

- I'd also suggest making the first part of the second sentence shorter. I suppose that the fact that this feature is a supported one should be clear. Not sure about this project, but on RHEL, unsupported features are usually listed under a separate section. You also already told the user in the first sentence and in the title that this feature is used to back up volumes, so what about dropping this part as well? So, what about just:

> `VolumePolicy` includes the `skip`, `snapshot`, and `fs-backup` actions.

Author: I agree, I've fixed it.
@mpershina, for your second point,the 'volumepolicy'  feature is the newly supported one instead of the previously used annotation method. Skip, fs-backup, and snapshot actions were supported by using annotation but now the process is streamlined. 
For the same reason, I've added the second point to explain what all is supported with the volumepolicy feature, and the same was verified by the SME

link:https://issues.redhat.com/browse/OADP-1071[OADP-1071]

> .File system backup and data mover can now use short term credentials

Reviewer 3: ```suggestion
.File system backup and data mover can now use short-term credentials
```

> File system backup and data mover can now use short term credentials such as AWS STS and GCP WIF. With this support, backup successfully completes without any `PartiallyFailed` status.

Reviewer 2: ```suggestion
File system backup and data mover can now use short-term credentials such as AWS STS and GCP WIF. With this support, backup successfully completes without any `PartiallyFailed` status.
```
* this is a compound adjective that modifies the following noun, use a hyphen.

link:https://issues.redhat.com/browse/OADP-5095[OADP-5095]

[id="resolved-issues-1-4-2_{context}"]
== Resolved issues

.DPA now reports errors if there is incorrect provider value in VSL 

> Previously, if the provider of a Volume Snapshot Location (VSL) spec was incorrect, Data Protection Application (DPA) reconciled successfully. With this update, DPA reports errors and requests for a valid provider value.

Reviewer 2: ```suggestion
Previously, if the provider of a Volume Snapshot Location (VSL) spec was incorrect, the Data Protection Application (DPA) reconciled successfully. With this update, DPA reports errors and requests for a valid provider value.
```

link:https://issues.redhat.com/browse/OADP-5044[OADP-5044]

.Data Mover restore is successful irrespective of using different OADP namespaces for backup and restore

> Previously, when backup operation was executed using OADP installed in one namespace but was restored using OADP installed in a different namespace, the Data Mover restore failed.

Reviewer 2: ```suggestion
Previously, when backup operation was executed using {oadp-short} installed in one namespace but was restored using {oadp-short} installed in a different namespace, the Data Mover restore failed.
```

Reviewer 3: Also, "by using" as per IBM SG (in both places).

> With this update, Data Mover restore is now successful.

Reviewer 2: Why is this on a different line. Looking at line 30, you have had this statement on the same line

Reviewer 3: + 1. Better to be consistent. Vote for moving everything under s single paragraph.

Author: fixed, thanks

link:https://issues.redhat.com/browse/OADP-5460[OADP-5460]

.SSE-C backup works with the calculated MD5 of the secret key

Previously, backup failed with the following error:
[source,terminal]
----
Requests specifying Server Side Encryption with Customer provided keys must provide the client calculated MD5 of the secret key.
----

> With this update, missing Server Side Encryption (SSE-C) base64 and MD5 hash is now fixed. As a result, SSE-C backup works with the calculated MD5 of the secret key. Along with this, incorrect `errorhandling` for the `customerKey` size is also fixed.

Reviewer 2: ```suggestion
With this update, missing Server-Side Encryption (SSE-C) base64 and MD5 hash are now fixed. As a result, SSE-C backup works with the calculated MD5 of the secret key. Along with this, incorrect `errorhandling` for the `customerKey` size is also fixed.
```

link:https://issues.redhat.com/browse/OADP-5388[OADP-5388]

For a complete list of all issues resolved in this release, see the list of link:https://issues.redhat.com/issues/?filter=12452476[OADP 1.4.2 resolved issues] in Jira.

[id="known-issues-1-4-2_{context}"]
== Known issues

.The nodeSelector spec is not supported for the Data Mover restore action

When a Data Protection Application (DPA) is created with the `nodeSelector` field set in the `nodeAgent` parameter, Data Mover restore partially fails instead of completing the restore operation. 
link:https://issues.redhat.com/browse/OADP-5260[OADP-5260]

.The S3 storage does not use proxy environment when tls skip verify is specified

In the image registry backup, the S3 storage does not use the proxy environment when the `insecureSkipTLSVerify` parameter is set to `true`.
link:https://issues.redhat.com/browse/OADP-3143[OADP-3143]

.Kopia does not delete the artifacts after backup expiration

Even after deleting a backup, Kopia does not delete the volume artifacts from the `${bucket_name}/kopia/${namespace}` on the S3 location after backup expired. 

> For more information, see link:https://docs.openshift.com/container-platform/4.17/backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-deleting-backups.html#oadp-about-kopia-repo-maintenance_deleting-backups[About Kopia repository maintenance].

Reviewer 2: Use an xref not a link.
